src/registrar/utility/csv_export.py

```diff
@@ -189,6 +197,82 @@ def write_csv(
     writer.writerows(total_body_rows)
 
 
+def get_requests(filter_condition, sort_fields):
+    """
+    Returns DomainRequest objects filtered and sorted based on the provided conditions.
+    filter_condition -> A dictionary of conditions to filter the objects.
+    sort_fields -> A list of fields to sort the resulting query set.
+    returns: A queryset of DomainRequest objects
+    """
+    requests = DomainRequest.objects.filter(**filter_condition).order_by(*sort_fields).distinct()
+    return requests
+
+
+def parse_request_row(columns, request: DomainRequest):
+    """Given a set of columns, generate a new row from cleaned column data"""
+
+    requested_domain_name = "No requested domain"
+
+    if request.requested_domain is not None:
+        requested_domain_name = request.requested_domain.name
+
+    if request.federal_type:
+        request_type = f"{request.get_organization_type_display()} - {request.get_federal_type_display()}"
+    else:
+        request_type = request.get_organization_type_display()
+
+    # create a dictionary of fields which can be included in output
+    FIELDS = {
+        "Requested domain": requested_domain_name,
+        "Status": request.get_status_display(),
+        "Organization type": request_type,
+        "Agency": request.federal_agency,
+        "Organization name": request.organization_name,
+        "City": request.city,
+        "State": request.state_territory,
+        "AO email": request.authorizing_official.email if request.authorizing_official else " ",
+        "Security contact email": request,
+        "Created at": request.created_at,
+        "Submission date": request.submission_date,
+    }
+
+    row = [FIELDS.get(column, "") for column in columns]
+    return row
+
+
+def write_requests_csv(
+    writer,
+    columns,
+    sort_fields,
+    filter_condition,
+    should_write_header=True,
+):
+    """Receives params from the parent methods and outputs a CSV with filtered and sorted requests.
+    Works with write_header as long as the same writer object is passed."""
+
+    all_requests = get_requests(filter_condition, sort_fields)
+
+    # Reduce the memory overhead when performing the write operation
+    paginator = Paginator(all_requests, 1000)
+
+    for page_num in paginator.page_range:
+        page = paginator.page(page_num)
+        rows = []
```

Comment:
@rachidatecs this will make the rows reset to empty on the next pagination and then when you go to make the report you will just get the last pagination because everything else ahas been overwritten. This was just a bug in the code for the domain csv so I think it just got copied here by mistake